### PR TITLE
Add top level documentation for Passkeys and Passkeys::Controllers

### PR DESCRIPTION
**Pull request by heliocola**

### CODE REVIEW

**Comment by heliocola** [comment]
@tcannonfodder : these are the 2 docs I was working on for a while.
I am pushing to get your feedback and see if it helps.

Please feel free to make changes to merge, if you want to go that way.
I will focus on the CSRF issue.

File: lib/devise/passkeys/controllers.rb [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
   module Passkeys
+    # Passkeys Controllers encapsulate all methods that implement actions related to the Passkeys
+    #   The most common uses cases are: registration, authentication, and reauthentication
+    #   Other controllers and concerns should all inherit from here. As the following:
+    #     ReauthenticationChallenge concern: provide methods for reauthentication challenge
+    #     Reauthentication concern: provide methods for reauthentication
+    #     PasskeysControllerConcern: provide methods to manage passkeys
+    #     ReauthenticationControllerConcern: provide methods for handling reauthentication
+    #     RegistrationsControllerConcern: provide methods for handling registration
+    #     SessionsControllerConcern: provide methods for handing session
```
Comment (heliocola) [inline lib/devise/passkeys/controllers.rb:20]: @tcannonfodder : if I remember correctly, our conversation, was leading into having a summary of these here, but it seems duplicated with the more detailed documentation in the module itself.

But perhaps here and/or inside the module itself where we should indicate the difference between the concerns inside the `concerns` folder vs the others.
Ref: https://github.com/ruby-passkeys/devise-passkeys/issues/4#issuecomment-1590357907

Comment (tcannonfodder) [reply lib/devise/passkeys/controllers.rb:20]: Yes, I agree!

**Comment by tcannonfodder** [comment]
Thanks so much for this! I'll do the editing pass on it, then get it merged in :)

**Review by tcannonfodder — APPROVED** [review]
Did an editing pass, making the `Devise::Passkeys` top-level documentation a signpost, and adding example code in the `Devise::Passkeys::Controllers` documentation

**Pull request merged**